### PR TITLE
docs(codepush): update hotfix docs for codepush

### PR DESCRIPTION
**Pull request by brainbicycle**
This PR resolves [PHIRE-403] 

### Description

Updates the hot fix docs to account for new codepush process.
Adds scripts for promoting releases and updating rollouts. 

### Follow-ups

- Suggestion from @joeyAghion would be nice to check if native code changed and fail deploys if so

### PR Checklist

- [ ] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- Add docs for codepush hot fix process - Brian 

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[PHIRE-403]: https://artsyproduct.atlassian.net/browse/PHIRE-403?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Dev changes (Add docs for codepush hot fix process - Brian)
- undefined (changes)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 61e43db56a54ca6df0aec0151c81c005245022b4
</p>

File: fastlane/Fastfile [hunk 1]
```diff
@@ -373,6 +373,22 @@ lane :deploy_to_codepush do |options|
...
+
+  sh("appcenter login --token #{token}")
+  puts "Here is where I should promote the codepush release"
+
+  sourceDeployment = "Canary"
+  targetDeployment = "HotfixTest"
+
+  # Promote to prod with 10% rollout, make it mandatory
+  sh("appcenter codepush promote -a #{androidAppName} -s #{sourceDeployment} -d #{targetDeployment} -r 10 --mandatory")
+  sh("appcenter codepush promote -a #{iosAppName} -s #{sourceDeployment} -d #{targetDeployment} -r 10 --mandatory")
```
Comment (gkartalis) [inline fastlane/Fastfile:389]: **question:** should we pass that 10% as a script variable?

Comment (brainbicycle) [reply fastlane/Fastfile]: done!

**Review by gkartalis — APPROVED** [review]
awesome!

**Pull request merged**